[WIP] feat: add lambda auth docs (#3429)
aws-amplify/docs: docs/lib/graphqlapi/fragments/native_common/authz/common.md

If you are using a 3rd party OIDC provider you will need to configure it and manage the details of token refreshes yourself. 

> ### Lambda

Reviewer: ```suggestion
### AWS Lambda
```

> You can implement your own API authorization logic using an AWS Lambda function. To add a Lambda authorizer as an authentication mode for you API, please see the Settings section of the AppSync console. You will need to manage the details of token refreshes in your application code yourself. 

Reviewer: ```suggestion
You can implement your own custom API authorization logic using an AWS Lambda function. To add a Lambda function as an authorization mode for your AppSync API, go to the **Settings** section of the **AppSync console**.

You will need to manage the details of token refreshes in your application code yourself. 
```

> 

Reviewer 2: should we also add a code snippet of an example of Lambda here?

Author: My plan was to link to the AppSync docs for Lambda auth here (once they become publicly available) and those would give customers more info on configuring a Lambda authorizer. We'll be adding CLI support in the near future, which would generate a basic authorizer for them.

<inline-fragment platform="ios" src="~/lib/graphqlapi/fragments/ios/authz/21_oidc.md"></inline-fragment>
<inline-fragment platform="android" src="~/lib/graphqlapi/fragments/android/authz/21_oidc.md"></inline-fragment>

> ### Lambda

Reviewer: ```suggestion
### AWS Lambda
```

> Update the `amplifyconfiguration.json` file and code snippet as follows:

Reviewer 2: nit: can we maybe rephrase this to point out that devs only need to update the `authorizationType` key?

> Update the `authorizationType` value in the `amplifyconfiguration.json` file and code snippet as follows:

Reviewer: ```suggestion
Amplify CLI does not currently allow you to configure Lambda as an authorization mode for your GraphQL API. To add a Lambda function as an authorization mode for your AppSync API, go to the **Settings** section of the **AppSync console**. Then, update the `authorizationType` value in the `amplifyconfiguration.json` file and code snippet as follows:
```
